Firestore/core/src/firebase/firestore/util/status.cc

```diff
@@ -1,28 +1,30 @@
-/* Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+/*
+ * Copyright 2015, 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
+#include "Firestore/core/src/firebase/firestore/util/status.h"
 
-    http://www.apache.org/licenses/LICENSE-2.0
+namespace firebase {
+namespace firestore {
+namespace util {
 
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-==============================================================================*/
-
-#include "tensorflow/core/lib/core/status.h"
-#include <stdio.h>
-
-namespace tensorflow {
-
-Status::Status(tensorflow::error::Code code, StringPiece msg) {
-  assert(code != tensorflow::error::OK);
+Status::Status(FirestoreErrorCode code, absl::string_view msg) {
+  FIREBASE_ASSERT(code != FirestoreErrorCode::Ok);
   state_ = std::unique_ptr<State>(new State);
   state_->code = code;
-  state_->msg = msg.ToString();
+  state_->msg = std::string(msg.data(), msg.data() + msg.size());
```

Comment:
Somewhat hilariously, this is `static_cast<std::string>(msg)` (`absl::string_view` has an explicit conversion operator).

Reply:
Done